bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -5,37 +5,29 @@ import java.io.IOException
 /**
  * Capture and serialize the state of all threads at the time of an exception.
  */
-internal class ThreadState : JsonStream.Streamable {
-
-    private val threads: List<Thread>
-
-    constructor(threads: List<Thread>) {
-        this.threads = threads
-    }
-
-    constructor(
-        config: ImmutableConfig,
-        currentThread: java.lang.Thread,
-        stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>>,
-        exc: Throwable?
-    ) {
+internal class ThreadState @JvmOverloads constructor
+    (
+    config: ImmutableConfig,
+    excStacktrace: Array<StackTraceElement>,
+    currentThread: java.lang.Thread = java.lang.Thread.currentThread(),
```

Comment:
Does this need a default? When wouldn't we specify one?

Reply:
Having this as a parameter is useful primarily for testing - for example, we use it within `ThreadStateTest` to verify that a different thread can be set as the 'errorReportingThread': https://github.com/bugsnag/bugsnag-android/pull/668/files#diff-5957b89ae75ed52bfc6a969a2a5ed6adR43

Otherwise, the report generation always takes place on the thread an exception was delivered on, so the value of the `currentThread` parameter should always be equivalent to `Thread.currentThread()`